src/v3/src/components/PasswordWithConfirmation/PasswordWithConfirmation.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box, FormHelperText } from '@mui/material';
+import { PasswordInput } from '@okta/odyssey-react-mui';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+
+import { getMessage } from '../../../../v2/ion/i18nTransformer';
+import { useWidgetContext } from '../../contexts';
+import { useValue } from '../../hooks';
+import {
+  ChangeEvent,
+  DataSchema,
+  FormBag,
+  PasswordWithConfirmationElement,
+  UISchemaElementComponent,
+} from '../../types';
+import InputPassword from '../InputPassword';
+
+const PasswordWithConfirmation: UISchemaElementComponent<{
+  uischema: PasswordWithConfirmationElement
+}> = ({ uischema }) => {
+  const { input, fieldRequiredErrorMessage, passwordMatchErrorMessage } = uischema.options;
+  const {
+    inputMeta: {
+      // @ts-ignore expose type from auth-js
+      messages: newPasswordMessages = {},
+    },
+  } = input.options;
+  const newPasswordError = newPasswordMessages?.value?.[0]
+    && getMessage(newPasswordMessages.value[0]);
+
+  const value = useValue(input);
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+  const [confirmPassword, setConfirmPassword] = useState<string>();
+  const [confirmPasswordError, setConfirmPasswordError] = useState<string | undefined>(
+    fieldRequiredErrorMessage,
+  );
+  const { dataSchemaRef } = useWidgetContext();
+
+  const handleConfirmPasswordValidation = (password: string | undefined): void => {
+    if (!password) {
+      setConfirmPasswordError(fieldRequiredErrorMessage);
+      return;
+    }
+
+    if (value !== password) {
+      setConfirmPasswordError(passwordMatchErrorMessage);
+      return;
+    }
+
+    setConfirmPasswordError(undefined);
+  };
+
+  const updateDataSchemaValidation = (confirmPw?: string): void => {
+    dataSchemaRef.current![input.name] = {
+      validate: (data: FormBag['data']) => {
+        const newPw = data[input.name];
+        const comparisonPw = (confirmPw ?? confirmPassword);
+        if (!newPw) {
+          setIsTouched(true);
+          return { i18n: { key: 'model.validation.field.blank' } };
+        }
+
+        if (comparisonPw !== newPw) {
+          setIsTouched(true);
+          // Do not display error on new password field when confirm is missing
+          // but do not allow submission
+          return { i18n: { key: '' } };
```

Comment:
Yes it does (I will attach a new video) but basically this validate function first validates that the new password field has a value, if it does not, it displays the field blank message. If that field does have a value, this logic is then executed which compares the new password with the confirm password. If they are not equivalent then we dont want to display an error on the new password field (because at this point we know a value exists) but we need to prevent submission until the two values match so we send an empty key here to prevent that submission (and the confirm password field displays its own error message).